apps/dashboard/app/apps/ood_files_app.rb

```diff
@@ -18,8 +18,22 @@ def candidate_favorite_paths
   # a link to the user's home directory
   # returns an array of other paths provided as shortcuts to the user
   def favorite_paths
-    @favorite_paths ||= candidate_favorite_paths.select {|p|
-      p.directory? && p.readable? && p.executable?
+     @favorite_paths ||= candidate_favorite_paths.select {|p| p.instance_of?(FavoritePath) ? p.uri.directory? && p.uri.readable? && p.uri.executable? 
+      : p.directory? && p.readable? && p.executable?
```

Comment:
`FavoritePath#uri` no longer exists but is used here. Be careful to run the tests and fix before committing, as the tests should have caught this.

Now, the right fix is to add the methods I suggest to FavoritePath below, and then change

```ruby
def candidate_favorite_paths
  self.class.candidate_favorite_paths || []
end
```

to

```ruby
def candidate_favorite_paths
  (self.class.candidate_favorite_paths || []).map { |path| FavoritePath.build(path) }
end
```

That way you end up working with only FavoritePath objects, and then you simplify this method:

```ruby
@favorite_paths ||= candidate_favorite_paths.select {|p| p.path.directory? && p.path.readable? && p.path.executable? }
```

If you also define `FavoritePath#accessible?` to do `path.directory? && path.readable? && path.executable?` you can do the pithy

```ruby
def favorite_paths
  @favorite_paths ||= candidate_favorite_paths.select(&:accessible?)
end
```